source/core/Array.cpp

```diff
@@ -1351,44 +1351,52 @@ struct ArrayReshapeStruct : public VarArgInstruction
 class ArrayIterator
 {
  private:
-    ArrayDimensionVector  _indexStack;
-    IntIndex _indexDim;
-    IntIndex*  dimensions;
-    IntIndex  _rank;
     TypedArrayCoreRef _array;
+    IntIndex  _rank;
+    IntIndex*  _dimensionsLengths;
```

Comment:
Please use _dimensionLengths (singular first word) to be less awkward and match similar variable usage elsewhere.